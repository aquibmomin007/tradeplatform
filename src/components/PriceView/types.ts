export type PriceViewProps = {
    bids: string[][]
    asks: string[][]
}