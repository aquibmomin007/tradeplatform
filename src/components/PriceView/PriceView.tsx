import { Col, Row, Table, Typography } from "antd";
import { useTranslation } from "react-i18next";
import { createUseStyles } from "react-jss";
import { PriceViewProps } from "./types";

const { Title } = Typography;

const useStyles = createUseStyles({
    appTable: {
        "& tr > td": {
            padding: "8px 15px"
        }
    },
})


export const PriceView = ({bids, asks}: PriceViewProps) => {
    const { t } = useTranslation();
    const classes = useStyles();
    
    if (!bids.length || !asks.length) {
        return null
    }

    const columns = [
        {title: "Price", dataIndex: "price", key: "price"},
        {title: "Quantity", dataIndex: "quantity", key: "quantity"},
    ]

    const format = ((d: string[], i: number) => ({ key: i, price: d[0], quantity: d[1]}))
    const bidsData = bids.map(format)
    const asksData = asks.map(format)

    return (
        <Row gutter={24}>
            <Col span={12}>
                <Title level={4} className="text-center title-spacing pb-0 ">{t('priceview.bids')}</Title>
                <Table rowKey="key" className={classes.appTable} columns={columns} dataSource={bidsData} pagination={false} />
            </Col>
            <Col span={12}>
                <Title level={4} className="text-center title-spacing pb-0 ">{t('priceview.asks')}</Title>
                <Table rowKey="key" className={classes.appTable} columns={columns} dataSource={asksData} pagination={false} /> 
            </Col>
        </Row>
    )
}
