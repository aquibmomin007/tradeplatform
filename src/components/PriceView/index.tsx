import { PriceView } from './PriceView';
export default PriceView;