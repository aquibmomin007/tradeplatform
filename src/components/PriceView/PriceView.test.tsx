import { render, screen } from '@testing-library/react';
import { PriceView } from './PriceView';

jest.mock('react-i18next', () => ({
    useTranslation: () => ({t: (key: string) => key})
}));

jest.mock('websocket', () => {
  return {
    w3cwebsocket: jest.fn()
  }
})

global.matchMedia = global.matchMedia || function () {
  return {
    addListener: jest.fn(),
    removeListener: jest.fn(),
  };
};

describe('PriceView', () => {

    test('render nothing when no bids and asks', () => {
        const { container } = render(<PriceView bids={[]} asks={[]} />)
        expect(container).toBeEmptyDOMElement()
    })

    test('check header content for table', () => {
        render(<PriceView bids={[["1", "0.1"], ["2", "0.2"]]} asks={[["1.1", "0.11"], ["1.2", "0.12"]]} />)

        expect(screen.getAllByText("Price")).toHaveLength(2)
        expect(screen.getAllByText("Quantity")).toHaveLength(2)
    })

    test('match snapshot', () => {
        const {container} = render(<PriceView bids={[["1", "0.1"], ["2", "0.2"]]} asks={[["1.1", "0.11"], ["1.2", "0.12"]]} />)
        expect(container.firstChild).toMatchSnapshot()
    })
})