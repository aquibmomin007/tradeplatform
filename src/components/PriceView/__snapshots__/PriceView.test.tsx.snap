// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceView match snapshot 1`] = `
<div
  class="ant-row"
  style="margin-left: -12px; margin-right: -12px;"
>
  <div
    class="ant-col ant-col-12"
    style="padding-left: 12px; padding-right: 12px;"
  >
    <h4
      class="ant-typography text-center title-spacing pb-0 "
    >
      priceview.bids
    </h4>
    <div
      class="ant-table-wrapper appTable-0-2-1"
    >
      <div
        class="ant-spin-nested-loading"
      >
        <div
          class="ant-spin-container"
        >
          <div
            class="ant-table"
          >
            <div
              class="ant-table-container"
            >
              <div
                class="ant-table-content"
              >
                <table
                  style="table-layout: auto;"
                >
                  <colgroup />
                  <thead
                    class="ant-table-thead"
                  >
                    <tr>
                      <th
                        class="ant-table-cell"
                      >
                        Price
                      </th>
                      <th
                        class="ant-table-cell"
                      >
                        Quantity
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    class="ant-table-tbody"
                  >
                    <tr
                      class="ant-table-row ant-table-row-level-0"
                      data-row-key="0"
                    >
                      <td
                        class="ant-table-cell"
                      >
                        1
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        0.1
                      </td>
                    </tr>
                    <tr
                      class="ant-table-row ant-table-row-level-0"
                      data-row-key="1"
                    >
                      <td
                        class="ant-table-cell"
                      >
                        2
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        0.2
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-col ant-col-12"
    style="padding-left: 12px; padding-right: 12px;"
  >
    <h4
      class="ant-typography text-center title-spacing pb-0 "
    >
      priceview.asks
    </h4>
    <div
      class="ant-table-wrapper appTable-0-2-1"
    >
      <div
        class="ant-spin-nested-loading"
      >
        <div
          class="ant-spin-container"
        >
          <div
            class="ant-table"
          >
            <div
              class="ant-table-container"
            >
              <div
                class="ant-table-content"
              >
                <table
                  style="table-layout: auto;"
                >
                  <colgroup />
                  <thead
                    class="ant-table-thead"
                  >
                    <tr>
                      <th
                        class="ant-table-cell"
                      >
                        Price
                      </th>
                      <th
                        class="ant-table-cell"
                      >
                        Quantity
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    class="ant-table-tbody"
                  >
                    <tr
                      class="ant-table-row ant-table-row-level-0"
                      data-row-key="0"
                    >
                      <td
                        class="ant-table-cell"
                      >
                        1.1
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        0.11
                      </td>
                    </tr>
                    <tr
                      class="ant-table-row ant-table-row-level-0"
                      data-row-key="1"
                    >
                      <td
                        class="ant-table-cell"
                      >
                        1.2
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        0.12
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
